subsys/bluetooth/services/ras/rreq/ras_rreq.c

```diff
@@ -0,0 +1,723 @@
+		}
+
+		LOG_DBG("Get Ranging Data Success");
+
+		break;
+	}
+	case BT_RAS_RREQ_CP_STATE_ACK_RD_WRITTEN: {
+		__ASSERT_NO_MSG(rreq->on_demand_rd.data_get_in_progress);
+		rreq->cp.state = BT_RAS_RREQ_CP_STATE_NONE;
+		if (rsp_code != RASCP_RESPONSE_SUCCESS) {
+			LOG_DBG("ACK Ranging Data returned an error %d", rsp_code);
+		}
+
+		data_receive_finished(rreq);
+		break;
+	}
+	default:
+		LOG_DBG("Unexpected Response code received %d", rsp_code);
+		break;
+	}
+}
+
+static uint8_t ras_cp_notify_func(struct bt_conn *conn, struct bt_gatt_subscribe_params *params,
+				  const void *data, uint16_t length)
+{
+	struct bt_ras_rreq *rreq = ras_rreq_find(conn);
+
+	struct net_buf_simple rsp;
+	net_buf_simple_init_with_data(&rsp, (uint8_t *)data, length);
+
+	uint8_t opcode = net_buf_simple_pull_u8(&rsp);
+
+	switch (opcode) {
+	case RASCP_RSP_OPCODE_COMPLETE_RD_RSP: {
+		if (rsp.len != RASCP_RSP_OPCODE_COMPLETE_RD_RSP_LEN) {
+			LOG_DBG("RAS-CP Complete RD RSP incorrect length: %d", length);
+			break;
+		}
+
+		uint16_t ranging_counter = net_buf_simple_pull_le16(&rsp);
+		if (!rreq->on_demand_rd.data_get_in_progress ||
+		    rreq->on_demand_rd.counter_in_progress != ranging_counter) {
+			LOG_DBG("RAS-CP Complete RD RSP unexpected ranging counter %d",
+				ranging_counter);
+			break;
+		}
+
+		ack_ranging_data(rreq);
+		break;
+	}
+	case RASCP_RSP_OPCODE_RSP_CODE: {
+		if (rsp.len != RASCP_RSP_OPCODE_RSP_CODE_LEN) {
+			LOG_DBG("RAS-CP RSP Code incorrect length: %d", length);
+			break;
+		}
+
+		uint8_t rsp_code = net_buf_simple_pull_u8(&rsp);
+		handle_rsp_code(rsp_code, rreq);
+
+		break;
+	}
+	default: {
+		LOG_DBG("Unkown RAS-CP RSP opcode: %d", opcode);
+		break;
+	}
+	}
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static uint8_t ras_on_demand_ranging_data_notify_func(struct bt_conn *conn,
+						      struct bt_gatt_subscribe_params *params,
+						      const void *data, uint16_t length)
+{
+	LOG_DBG("On-demand Ranging Data notification received");
+
+	struct bt_ras_rreq *rreq = ras_rreq_find(conn);
+
+	if (!rreq->on_demand_rd.data_get_in_progress) {
+		LOG_DBG("Unexpected On-demand Ranging Data notification received");
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	if (length < 2) {
+		LOG_DBG("On-demand Ranging Data notification received invalid length");
+		rreq->on_demand_rd.error_with_data_receive = true;
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	if (rreq->on_demand_rd.last_segment_received) {
+		LOG_DBG("On-demand Ranging Data notification received after last segment");
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	if (rreq->on_demand_rd.error_with_data_receive) {
+		/* Already had an error receiving this ranging counter, so exit here. */
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	struct net_buf_simple segment;
+	net_buf_simple_init_with_data(&segment, (uint8_t *)data, length);
+
+	uint8_t segmentation_header = net_buf_simple_pull_u8(&segment);
+
+	bool first_segment = segmentation_header & BIT(0);
+	bool last_segment = segmentation_header & BIT(1);
```

Comment:
Might be nice to explicitly make this a bool
```suggestion
	bool last_segment = !!(segmentation_header & BIT(1));
```

Reply:
isn't it safe ? assigning/casting to bool

Reply:
Yup apparently with stdbool and _Bool this is fine, only matters for pre-c99 so disregard my comment